packages/react-magma-dom/src/components/CharacterCounter/CharacterCounter.test.js

```diff
@@ -0,0 +1,232 @@
+    );
+
+    fireEvent.change(getByLabelText(labelText), { target: { value: 'dddd' } });
+
+    expect(onChange).toHaveBeenCalled();
+    expect(getByText('0 ' + charactersLeft)).toBeInTheDocument();
+  });
+
+  it('Shows the label "character left" as the user types to one remaining character', () => {
+    const onChange = jest.fn();
+    const labelText = 'Character Counter';
+    const characterLeft = defaultI18n.characterCounter.characterLeft;
+    const { getByText, getByLabelText } = render(
+      <Input labelText={labelText} maxLength={4} onChange={onChange} />
+    );
+
+    fireEvent.change(getByLabelText(labelText), { target: { value: 'ddd' } });
+
+    expect(onChange).toHaveBeenCalled();
+    expect(getByText('1 ' + characterLeft)).toBeInTheDocument();
+  });
+
+  it('Shows the label "character over limit" as the user types over the limit by one', () => {
+    const onChange = jest.fn();
+    const labelText = 'Character Counter';
+    const characterOver = defaultI18n.characterCounter.characterOver;
+    const { getByText, getByLabelText } = render(
+      <Input labelText={labelText} maxLength={4} onChange={onChange} />
+    );
+
+    fireEvent.change(getByLabelText(labelText), { target: { value: 'ddddd' } });
+
+    expect(onChange).toHaveBeenCalled();
+
+    expect(getByText('1 ' + characterOver)).toHaveStyleRule(
+      'color',
+      magma.colors.danger
+    );
+    expect(getByText('1 ' + characterOver)).toBeInTheDocument();
+  });
+
+  it('Shows the label "characters over limit" as the user types over the limit', () => {
+    const onChange = jest.fn();
+    const labelText = 'Character Counter';
+    const charactersOver = defaultI18n.characterCounter.charactersOver;
+    const { getByText, getByLabelText } = render(
+      <Input labelText={labelText} maxLength={4} onChange={onChange} />
+    );
+
+    fireEvent.change(getByLabelText(labelText), {
+      target: { value: 'dddddd' },
+    });
+
+    expect(onChange).toHaveBeenCalled();
+    expect(getByText('2 ' + charactersOver)).toBeInTheDocument();
+  });
+
+  it('Shows the inverse label "characters over limit" as the user types over the limit', () => {
+    const onChange = jest.fn();
+    const labelText = 'Character Counter';
+    const charactersOver = defaultI18n.characterCounter.charactersOver;
+    const { getByText, getByLabelText } = render(
+      <Input
+        isInverse
+        labelText={labelText}
+        maxLength={4}
+        onChange={onChange}
+      />
+    );
+
+    fireEvent.change(getByLabelText(labelText), {
+      target: { value: 'dddddd' },
+    });
+
+    expect(onChange).toHaveBeenCalled();
+    expect(getByText('2 ' + charactersOver)).toHaveStyleRule(
+      'color',
+      magma.colors.danger200
+    );
+
+    expect(getByText('2 ' + charactersOver)).toBeInTheDocument();
+  });
+
+  it('Shows the label "characters left" equal to the maxLength if the user clears the input', () => {
```

Comment:
I think it would help with legibility of tests (and so that we can ensure we've covered every case we need to) to group tests based on characterLeft, characterOver, etc